### PR TITLE
Force App Update

### PR DESCRIPTION
**Pull request by cristhianescobar**
- Uptake new Config API changes
- Added the hook to ensure we can force upgrade from login activity or home page

### CODE REVIEW

File: android/app-newm/build.gradle.kts [hunk 1]
```diff
@@ -24,7 +24,7 @@ android {
         minSdk = libs.versions.android.minSdk.get().toInt()
         targetSdk = libs.versions.android.targetSdk.get().toInt()
         versionCode = 5
```
Comment (newmskywalker) [inline android/app-newm/build.gradle.kts:26]: should we be bumping the versionCode as well? 
Maybe we can automate it in a future pr

File: shared/src/commonMain/kotlin/io.newm.shared/internal/implementations/ForceAppUpdateUseCaseImpl.kt [hunk 2]
```diff
@@ -1,27 +1,35 @@
...
 internal class ForceAppUpdateUseCaseImpl(
     private val remoteConfigRepository: RemoteConfigRepository,
-    private val sharedBuildConfig: NewmSharedBuildConfig
 ) : ForceAppUpdateUseCase {
 
-    override suspend fun isAndroidUpdateRequired(): Boolean =
-        shouldForceAppUpdate { it.android.minAppVersion }
+    override suspend fun isAndroidUpdateRequired(
+        currentAppVersion: String,
+        humanVerificationCode: String
```
Comment (newmskywalker) [inline shared/src/commonMain/kotlin/io.newm.shared/internal/implementations/ForceAppUpdateUseCaseImpl.kt:14]: Are we going to need a captcha for all requests in the future? If so, perhaps we should start thinking about adding the headers using a request interceptor

**Review by newmskywalker — APPROVED** [review]
lgtm

File: shared/src/commonMain/kotlin/io.newm.shared/internal/implementations/ForceAppUpdateUseCaseImpl.kt (hunk 2 above)
Comment (cristhianescobar) [reply shared/src/commonMain/kotlin/io.newm.shared/internal/implementations/ForceAppUpdateUseCaseImpl.kt:14]: Up until now we had used captcha for prelogin request, this is the first time we use it on post login flows. We can check with Walter what the roadmap looks like for BE.

File: android/app-newm/build.gradle.kts (hunk 1 above)
Comment (cristhianescobar) [reply android/app-newm/build.gradle.kts:26]: Yeah we probably need to automate both version code and string, we can do that on another PR

**Pull request merged**